zerver/lib/zcommand.py

```diff
@@ -73,4 +77,52 @@ def change_mode_setting(
                 setting_value=False,
             )
         )
+    elif command == "digress":
+        data_keys = [
+            "old_stream",
+            "old_topic",
+            "new_stream",
+            "new_topic",
+        ]
+        if not data or not all(key in data for key in data_keys):
+            raise JsonableError(_("Invalid data."))
+
+        if data["old_stream"] == data["new_stream"] and data["old_topic"] == data["new_topic"]:
+            return dict(msg=_("Cannot digress to the same thread."))
+
+        (old_stream, ignored_old_stream_sub) = access_stream_by_name(
+            user_profile, data["old_stream"]
+        )
+
+        (new_stream, ignored_new_stream_sub) = access_stream_by_name(
+            user_profile, data["new_stream"]
+        )
+
+        old_topic_exists = user_message_exists_for_topic(
+            user_profile=user_profile,
+            recipient_id=old_stream.recipient_id,
+            topic_name=data["old_topic"],
+        )
+
+        if not old_topic_exists:
+            return dict(
+                msg=_("Old topic #**{}>{}** does not exist.").format(
+                    data["old_stream"], data["old_topic"]
+                )
```

Comment:
It's a little awkward that this is a `json_success` -- shouldn't we `raise JsonableError` here instead, since the user's command didn't work?

(I'm not super familiar with the slash command system's error handling conventions, so this comment could be wrong).

Reply:
Yes, makes sense. I changed all these to raise `JsonableError`. These will be shown above the compose box (in the second commit).

All slash commands currently send `success`, and the messages are latter shown in the feedback widget. I think that is because the current slash commands are simple enough that they can't really "fail".